### PR TITLE
Integration of Californium 2.0.0 M8

### PR DESCRIPTION
**Pull request by sbernard31**
Aims to solve issue about #476

### CODE REVIEW

File: leshan-server-cluster/src/main/java/org/eclipse/leshan/server/cluster/RedisRegistrationStore.java [hunk 1]
```diff
@@ -499,9 +499,11 @@ public Observation getObservation(String registrationId, byte[] observationId) {
                 byte[] serializeObs = serializeObs(obs);
                 byte[] previousValue = null;
                 if (ifAbsent) {
-                    if (j.setnx(key, serializeObs) == 0) {
-                        // not set => already exist
-                        previousValue = j.get(key);
+                    previousValue = j.get(key);
+                    if (previousValue == null || previousValue.length > 0) {
```
Comment (boaks) [inline leshan-server-cluster/src/main/java/org/eclipse/leshan/server/cluster/RedisRegistrationStore.java]: are you sure `previousValue.length > 0`?
I think it should be ==

**Comment by boaks** [comment]
Very good. Can I close my old PR? It doesn't make sense, if you complete it here.

**Comment by sbernard31** [comment]
I close the old PR.
@boaks does this PR sounds good to you ? Should I merge it ?

File: leshan-server-cluster/src/main/java/org/eclipse/leshan/server/cluster/RedisRegistrationStore.java [hunk 2]
```diff
@@ -482,16 +495,26 @@ public void add(org.eclipse.californium.core.observe.Observation obs) {
...
                 if (!j.exists(toRegIdKey(registrationId)))
                     throw new IllegalStateException("no registration for this Id");
-
-                byte[] previousValue = j.getSet(toKey(OBS_TKN, obs.getRequest().getToken()), serializeObs(obs));
+                byte[] key = toKey(OBS_TKN, obs.getRequest().getToken().getBytes());
+                byte[] serializeObs = serializeObs(obs);
+                byte[] previousValue = null;
+                if (ifAbsent) {
+                    previousValue = j.get(key);
+                    if (previousValue == null || previousValue.length > 0) {
```
Comment (boaks) [inline leshan-server-cluster/src/main/java/org/eclipse/leshan/server/cluster/RedisRegistrationStore.java]: I still have concerns, if `previousValue.length > 0` is correct, I think it must be `previousValue.length == 0`.

Comment (sbernard31) [reply leshan-server-cluster/src/main/java/org/eclipse/leshan/server/cluster/RedisRegistrationStore.java]: You're right ! I will fix that.

**Pull request merged**